pkg/controller/datavolume/import-controller.go

```diff
@@ -434,3 +502,82 @@ func (r *ImportReconciler) getNextCheckpoint(dataVolume *cdiv1.DataVolume, pvc *
 
 	return nil
 }
+
+func volumeImportSourceName(dv *cdiv1.DataVolume) string {
+	return fmt.Sprintf("%s-%s", volumeImportSourcePrefix, dv.UID)
+}
+
+func (r *ImportReconciler) createVolumeImportSourceCR(syncState *dvSyncState) error {
+	dv := syncState.dvMutated
+	importSource := &cdiv1.VolumeImportSource{}
+	importSourceName := volumeImportSourceName(dv)
+
+	exists, err := cc.GetResource(context.TODO(), r.client, dv.Namespace, importSourceName, importSource)
+	if err != nil {
+		return err
+	}
+	if exists {
+		// import source already exists
+		return nil
+	}
+
+	source := &cdiv1.ImportSourceType{}
+	if http := dv.Spec.Source.HTTP; http != nil {
+		source.HTTP = http
+	} else if s3 := dv.Spec.Source.S3; s3 != nil {
+		source.S3 = s3
+	} else if gcs := dv.Spec.Source.GCS; gcs != nil {
+		source.GCS = gcs
+	} else if registry := dv.Spec.Source.Registry; registry != nil {
+		source.Registry = registry
+	} else if imageio := dv.Spec.Source.Imageio; imageio != nil {
+		source.Imageio = imageio
+	} else if vddk := dv.Spec.Source.VDDK; vddk != nil {
+		source.VDDK = vddk
+	} else {
+		// Our dv shouldn't be without source
+		// Defaulting to Blank source
+		source.Blank = &cdiv1.DataVolumeBlankImage{}
+	}
+
+	importSource = &cdiv1.VolumeImportSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      importSourceName,
+			Namespace: dv.Namespace,
+		},
+		Spec: cdiv1.VolumeImportSourceSpec{
+			Source:        source,
+			ContentType:   dv.Spec.ContentType,
+			Preallocation: dv.Spec.Preallocation,
+		},
```

Comment:
Is this getting owned by the DV so that it will be cleaned up in case the DV is deleted before population succeeds?

Reply:
I believe so does the PVC and no need to the import source if no pvc to populate